src/resolve/resolved-dns-rr.c

```diff
@@ -826,6 +827,115 @@ static char *format_txt(DnsTxtItem *first) {
+                                return NULL;
+                        strv_push(&values_strv, alpn);
+                        offset += sz;
+                }
+                value = strv_join(values_strv, ",");
+                break;
+        }
+        case DNS_SVC_PARAM_KEY_PORT: {
+                uint16_t port = unaligned_read_be16(i->value);
+                r = asprintf(&value, "%" PRIu16, port);
+                if (r < 0)
+                        return NULL;
+                break;
+        }
+        case DNS_SVC_PARAM_KEY_IPV4HINT: {
+                const struct in_addr *addrs = (struct in_addr*) &i->value;
+                _cleanup_strv_free_ char **values_strv = NULL;
+                for (size_t n = 0; n < i->length / sizeof (struct in_addr); n++) {
+                        char *addr;
+                        r = in_addr_to_string(AF_INET, (const union in_addr_union*) &addrs[n], &addr);
+                        if (r < 0)
+                                return NULL;
+                        strv_push(&values_strv, addr);
+                }
+                value = strv_join(values_strv, ",");
+                if (!value)
+                        return NULL;
+                break;
+        }
+        case DNS_SVC_PARAM_KEY_IPV6HINT: {
+                const struct in6_addr *addrs = (struct in6_addr*) &i->value;
+                _cleanup_strv_free_ char **values_strv = NULL;
+                for (size_t n = 0; n < i->length / sizeof (struct in6_addr); n++) {
+                        char *addr;
+                        r = in_addr_to_string(AF_INET6, (const union in_addr_union*) &addrs[n], &addr);
+                        if (r < 0)
+                                return NULL;
+                        strv_push(&values_strv, addr);
+                }
+                value = strv_join(values_strv, ",");
+                if (!value)
+                        return NULL;
+                break;
+        }
+        default: {
+                char *p;
+                p = value = new(char, 4 * i->length + 3);
+                *(p++) = '"';
+                for (size_t j = 0; j < i->length; j++) {
+                        if (i->value[j] < ' ' || IN_SET(i->value[j], ',', '\\', '"') ||
+                            i->value[j] >= 127) {
+                                *(p++) = '\\';
+                                *(p++) = '0' + (i->value[j] / 100);
+                                *(p++) = '0' + ((i->value[j] / 10) % 10);
+                                *(p++) = '0' + (i->value[j] % 10);
+                        } else
+                                *(p++) = i->value[j];
+                }
+                *(p++) = '"';
+                *p = '\0';
+                break;
+        }
+        }
+
+        return value;
+}
+
+static char *format_svc_param(DnsSvcParam *i) {
+        char *s;
+        const char *key = FORMAT_DNS_SVC_PARAM_KEY(i->key);
+
+        _cleanup_free_ char *value = NULL;
+
+        if (!i->length) {
```

Comment:
as per coding style, for integer values that are used as numerics (i.e. not just as bools) we want explicit zero/non-zero comparisons, hence please write `if (i->length == 0)`